File: app/src/main/java/com/example/breezil/favnews/di/AppComponent.java
package com.example.breezil.favnews.di;

import android.app.Application;

import com.example.breezil.favnews.FavNewsApp;
import com.example.breezil.favnews.di.module.AppModule;
import com.example.breezil.favnews.di.module.ArticleDetailActivityModule;
import com.example.breezil.favnews.di.module.MainActivityModule;
import com.example.breezil.favnews.di.module.SearchActivityModule;
import com.example.breezil.favnews.di.module.SettingsActivityModule;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjectionModule;

@Singleton
@Component(modules = {
        AndroidInjectionModule.class,
        AppModule.class,
        MainActivityModule.class,
        ArticleDetailActivityModule.class,
        SearchActivityModule.class,
        SettingsActivityModule.class
})
public interface AppComponent {
    @Component.Builder
    interface Builder {
        @BindsInstance
        Builder application(Application application);

        AppComponent build();
    }

    void inject(FavNewsApp favNewsApp);
}

File: app/src/main/java/com/example/breezil/fave/di/AppComponent.kt
package com.example.breezil.fave.di

import android.app.Application

import com.example.breezil.fave.Fave
import com.example.breezil.fave.di.module.AppModule
import com.example.breezil.fave.di.module.MainActivityModule
import com.example.breezil.fave.di.module.SearchActivityModule
import com.example.breezil.fave.di.module.SettingsActivityModule

import javax.inject.Singleton

import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjectionModule

@Singleton
@Component(modules = [AndroidInjectionModule::class,
    AppModule::class,
    MainActivityModule::class,
    SearchActivityModule::class,
    SettingsActivityModule::class])
interface AppComponent {
    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: Application): Builder

        fun build(): AppComponent
    }

    fun inject(fave: Fave)
}
